src/integrations/watchers/indexable-post-type-change-watcher.php

```diff
@@ -99,7 +99,9 @@ public function check_post_types_public_availability() {
 			return;
 		}
 
-		$public_post_types            = \array_keys( $this->post_type_helper->get_public_post_types() );
+		$excluded_post_types = $this->post_type_helper->get_excluded_post_types_for_indexables();
+		$public_post_types   = \array_diff( \array_keys( $this->post_type_helper->get_public_post_types() ), $excluded_post_types );
```

Comment:
You should use the `get_indexable_post_types` here instead of doing these checks. Right now this would miss post types excluded in `filter_included_post_types`